lib/puppet/type/iis_site.rb

```diff
@@ -0,0 +1,270 @@
+require 'puppet/parameter/boolean'
+
+Puppet::Type.newtype(:iis_site) do
+  @doc = "Create a new IIS website."
+
+  newproperty(:ensure) do
+    desc "Specifies whether a site should be present or absent. If present is
+      specified, the site will be created but left in the default stopped state.
+      If started is specified, then the site will be created as well as started.
+      If stopped is specified, then the site will be created and kept stopped."
```

Comment:
This is interesting - so a user could specify `ensure => present` in their manifest and the site could actually be in either the `started` or `stopped` state? (ie if the user manually started the site after puppet ran with `ensure => present`